source/server/hot_restarting_base.h

```diff
@@ -27,73 +27,76 @@ namespace Server {
  */
 class HotRestartingBase : public Logger::Loggable<Logger::Id::main> {
 protected:
-  HotRestartingBase(uint64_t base_id) : base_id_(base_id) {}
-  ~HotRestartingBase();
-
-  void initDomainSocketAddress(sockaddr_un* address);
-  sockaddr_un createDomainSocketAddress(uint64_t id, const std::string& role,
-                                        const std::string& socket_path, mode_t socket_mode);
-  void bindDomainSocket(uint64_t id, const std::string& role, const std::string& socket_path,
-                        mode_t socket_mode);
-  int myDomainSocket() const { return my_domain_socket_; }
-
-  // Protocol description:
-  //
-  // In each direction between parent<-->child, a series of pairs of:
-  //   A uint64 'length' (bytes in network order),
-  //   followed by 'length' bytes of a serialized HotRestartMessage.
-  // Each new message must start in a new sendmsg datagram, i.e. 'length' must always start at byte
-  // 0. Each sendmsg datagram can be up to 4096 bytes (including 'length' if present). When the
-  // serialized protobuf is longer than 4096-8 bytes, and so cannot fit in just one datagram, it is
-  // delivered by a series of datagrams. In each of these continuation datagrams, the protobuf data
-  // starts at byte 0.
-  //
-  // There is no mechanism to explicitly pair responses to requests. However, the child initiates
-  // all exchanges, and blocks until a reply is received, so there is implicit pairing.
-  void sendHotRestartMessage(sockaddr_un& address, const envoy::HotRestartMessage& proto);
+  HotRestartingBase(uint64_t base_id) : main_rpc_stream_(base_id) {}
 
   enum class Blocking { Yes, No };
-  // Receive data, possibly enough to build one of our protocol messages.
-  // If block is true, blocks until a full protocol message is available.
-  // If block is false, returns nullptr if we run out of data to receive before a full protocol
-  // message is available. In either case, the HotRestartingBase may end up buffering some data for
-  // the next protocol message, even if the function returns a protobuf.
-  std::unique_ptr<envoy::HotRestartMessage> receiveHotRestartMessage(Blocking block);
-
-  bool replyIsExpectedType(const envoy::HotRestartMessage* proto,
-                           envoy::HotRestartMessage::Reply::ReplyCase oneof_type) const;
 
   // Returns a Gauge that tracks hot-restart generation, where every successive
   // child increments this number.
   static Stats::Gauge& hotRestartGeneration(Stats::Scope& scope);
```

Comment:
This is the only method remaining in HotRestartingBase. Should we just rename HotRestartingBase to RpcStream and move this method to a util function? 

On the other hand, the reason to extract RpcStream is because HotRestartingBase has code specific to TCP. But it seems that almost all the code will be reused.

Reply:
It's not just the methods, it's the protected members too. After this change, that's just the one rpc_stream, but with the following change it's both rpc_streams. To some extent it might make sense to refactor that significantly and make the parent and child instances have totally distinct initializers, but I'm trying to make the *minimum* change to the existing structure that will allow the necessary functionality to be added - I'd rather not add massive cleanup time to my roster.

Reply:
Can you do s/HotRestartingBase/RpcStream in the .h and .cc file and add a class definition somewhere?

```
class HotRestartingBase {
protected:
RpcStream main_rpc_stream_;
}
```
At the first glance, this will do what the same thing as the current PR does and have a smaller diff for code review.

Reply:
Done.